flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/DataFormatConverters.java

```diff
@@ -0,0 +1,1022 @@
+		t2C.put(BasicTypeInfo.INT_TYPE_INFO, IntConverter.INSTANCE);
+		t2C.put(BasicTypeInfo.LONG_TYPE_INFO, LongConverter.INSTANCE);
+		t2C.put(BasicTypeInfo.FLOAT_TYPE_INFO, FloatConverter.INSTANCE);
+		t2C.put(BasicTypeInfo.DOUBLE_TYPE_INFO, DoubleConverter.INSTANCE);
+		t2C.put(BasicTypeInfo.SHORT_TYPE_INFO, ShortConverter.INSTANCE);
+		t2C.put(BasicTypeInfo.BYTE_TYPE_INFO, ByteConverter.INSTANCE);
+		t2C.put(BasicTypeInfo.CHAR_TYPE_INFO, CharConverter.INSTANCE);
+
+		t2C.put(PrimitiveArrayTypeInfo.BOOLEAN_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveBooleanArrayConverter.INSTANCE);
+		t2C.put(PrimitiveArrayTypeInfo.INT_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveIntArrayConverter.INSTANCE);
+		t2C.put(PrimitiveArrayTypeInfo.LONG_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveLongArrayConverter.INSTANCE);
+		t2C.put(PrimitiveArrayTypeInfo.FLOAT_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveFloatArrayConverter.INSTANCE);
+		t2C.put(PrimitiveArrayTypeInfo.DOUBLE_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveDoubleArrayConverter.INSTANCE);
+		t2C.put(PrimitiveArrayTypeInfo.SHORT_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveShortArrayConverter.INSTANCE);
+		t2C.put(PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveByteArrayConverter.INSTANCE);
+		t2C.put(PrimitiveArrayTypeInfo.CHAR_PRIMITIVE_ARRAY_TYPE_INFO, PrimitiveCharArrayConverter.INSTANCE);
+
+		t2C.put(SqlTimeTypeInfo.DATE, DateConverter.INSTANCE);
+		t2C.put(SqlTimeTypeInfo.TIME, TimeConverter.INSTANCE);
+		t2C.put(SqlTimeTypeInfo.TIMESTAMP, TimestampConverter.INSTANCE);
+
+		t2C.put(BinaryStringTypeInfo.INSTANCE, BinaryStringConverter.INSTANCE);
+
+		TYPE_INFO_TO_CONVERTER = Collections.unmodifiableMap(t2C);
+	}
+
+	public static DataFormatConverter getConverterForTypeInfo(TypeInformation typeInfo) {
+		DataFormatConverter converter = TYPE_INFO_TO_CONVERTER.get(typeInfo);
+		if (converter != null) {
+			return converter;
+		}
+
+		if (typeInfo instanceof BasicArrayTypeInfo) {
+			return new ObjectArrayConverter(((BasicArrayTypeInfo) typeInfo).getComponentInfo());
+		} else if (typeInfo instanceof ObjectArrayTypeInfo) {
+			return new ObjectArrayConverter(((ObjectArrayTypeInfo) typeInfo).getComponentInfo());
+		} else if (typeInfo instanceof MapTypeInfo) {
+			MapTypeInfo mapType = (MapTypeInfo) typeInfo;
+			return new MapConverter(mapType.getKeyTypeInfo(), mapType.getValueTypeInfo());
+		} else if (typeInfo instanceof RowTypeInfo) {
+			return new RowConverter((RowTypeInfo) typeInfo);
+		} else if (typeInfo instanceof PojoTypeInfo) {
+			return new PojoConverter((PojoTypeInfo) typeInfo);
+		} else if (typeInfo instanceof TupleTypeInfo) {
+			return new TupleConverter((TupleTypeInfo) typeInfo);
+		} else if (typeInfo instanceof TupleTypeInfoBase && Product.class.isAssignableFrom(typeInfo.getTypeClass())) {
+			return new CaseClassConverter((TupleTypeInfoBase) typeInfo);
+		} else if (typeInfo instanceof BinaryArrayTypeInfo) {
+			return BinaryArrayConverter.INSTANCE;
```

Comment:
Why do we need to convert between `BinaryArray` and `BinaryArrary`, same question to `BinaryMap` and `BinaryString`

Reply:
Some scenarios Udx needs to use BinaryString for performance, or put BinaryString in Row. These converters can handle these scenarios very well.